test/cache-tests.ts

```diff
@@ -234,18 +236,18 @@ describe('S3 tests', () => {
     basicTests(() => new S3Cache('test', 'test.bucket', 'cache', 'uk-north-1'));
 
     it('should correctly handle errors', () => {
-        let err = null;
-        const cache = new S3Cache('test', 'test.bucket', 'cache', 'uk-north-1', (e, op) => {
+        let err: Error | null = null;
+        const cache = new S3Cache('test', 'test.bucket', 'cache', 'uk-north-1', (e: Error, op) => {
             err = e;
             op.should.equal('read');
         });
-        throwS3Error = new Error('Some s3 error');
+        throwS3Error = new Error('Some s3 error') as AWSError;
         return cache
-            .get('doesntmatter', 'subsystem')
+            .get('doesntmatter')
             .should.eventually.contain({hit: false})
             .then(x => {
                 cache.stats().should.eql({hits: 0, puts: 0, gets: 1});
-                err.toString().should.equal('Error: Some s3 error');
+                err!.toString().should.equal('Error: Some s3 error');
```

Comment:
The GHA tests we run are saying "Forbidden non-null assertion" on this - I'm not sure we deliberately did anything here though. @jeremy-rifkin I've not seen `!` before - what's your thoughts (I know we have an `assert` around). I know this is in tests, so I'm less worried but in the interests of learning I'd like to know if `!` is good vs assert etc.

Reply:
TLDR my thoughts are it can be ok but may as well `unwrap` so we get a runtime check for it.

Typescript's `!` can be quick and helpful when you really know something is non-null but of course it's better either find a better way to type things or use something like unwrap in case anything ever goes wrong.

Reply:
In fairness here this is a test where it'd "just fail" in that case. I think I understand now. I'll merge as-is though, and we can discuss more :)